CMakeLists.txt

```diff
@@ -553,6 +553,15 @@ elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
             PROPERTIES COMPILE_OPTIONS "-Wno-unused-parameter"
         )
     endif()
+    # Compiler ID Clang and Frontend variant MSVC means we're using clang-cl
+    # Not sure why but "${CMAKE_CXX_COMPILER_FRONTEND_VARIANT}" doesn't
+    # evaluate in this if statement as expected
+    if (CMAKE_CXX_COMPILER_FRONTEND_VARIANT MATCHES "MSVC")
```

Comment:
I was seeing this across a couple clangcl versions. I have 12 installed locally and github actions has 13 IIRC. This didn't match any of the existing version checks. ClangCL also continues to be weird because none of the existing version checks trigger for my local build and the unused-function warning is not disabled on zstd.c with any of the existing checks. Unsure which versions this affects and if it would ever change and since this does only happen here with clang-cl, I checked for clang-cl explicitly. I could check for any version greater than or equal to 12.0.0 if you'd prefer

Reply:
Decided to test out building with regular clang on windows and realizing that these warning disables are still needed even without clang-cl specifically so I will update these

Reply:
Thanks.